### DISCUSSION TITLE
Discussion: ansible-core 2.17 removing target support for Python 2.7 and 3.6

### DISCUSSION THREAD

**Discussion by felixfontein** [ansible-collections/news-for-maintainers#55] [Discussions on announcements]
If someone wants to discuss this topic (#54), please do it here.

**Comment by Thulium-Drake** [comment]
I fear this might have some consequences for RHEL8 based systems, what is the suggested path to follow here? Can we 'just' upgrade to Python 3.6+ (3.9 or so) on RHEL8. I'm mainly looking for people's experiences :-)

As of now I and my colleagues 'just' use whatever RHEL8 comes with, but AFAIK the ```platform-python``` is 3.6.8, which will then break. And RHEL8 isn't going anywhere anytime soon.

**Reply by sivel** [reply]
We are still working out the details with the RHEL teams around what version of ansible-core RHEL will continue ship with.  As of now, it looks like RHEL8 will ship with ansible-core 2.16 for the remainder of it's lifetime, but only supported in the context of the RHEL system roles use cases.

**Reply by richm** [reply]
> only supported in the context of the RHEL system roles use cases

and other RHEL ansible content such as ansible-freeipa, ansible-pcp, SAP roles, etc. that are provided with RHEL as RHEL RPM packages

**Reply by Thulium-Drake** [reply]
Doesn't that also mean that RH is committing to keep ansible-core 2.16 around as somewhat of a LTS release? I get that you wont support community managed collections. But I would expect that everything that is covered in the AAP subscription and all RH-certified content will still be supported against RHEL8 targets (for as long as the vendor/supplier does that, which isn't unreasonable given the remaining lifetime of RHEL8).

And as RHEL8 is still around for another 8 years, that'd mean there's  still a legitimate reason for having Ansible support against targets with Python 3.6.8 and lower.

Please don't get me wrong, I really appreciate the work you all do. You have made my work and world so much more fun in the past years, but I'm worried this might upset _a lot_ of people who aren't ready to upgrade their entire fleets of RHEL systems to 9. And unless we get an upgraded version of ```platform-python``` in RHEL8, I'm not too sure if that's the direction the project needs to take.

**Reply by sivel** [reply]
> Doesn't that also mean that RH is committing to keep ansible-core 2.16 around as somewhat of a LTS release?

Yes, and no.

As explained above it will be supported in the context of the Red Hat maintained RHEL (and related) ansible content, for the lifecycle of RHEL8. This does not necessarily apply to AAP, as AAP has it's own independent lifecycles, separate of RHEL. No decisions have specifically been made in regards to AAP content yet, but it may potentially match the RHEL plans. Generally speaking, AAP has not supported OSes once they enter ELS.

It will not be a traditional LTS, as it won't receive bug fixes or new features. After upstream support ends as defined at https://docs.ansible.com/ansible/devel/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix, ansible-core 2.16 downstream will receive critical security fixes only, and only where it intersects functionality used by the above mentioned content.

**Reply by Thulium-Drake** [reply]
Gotcha, so basically the same as the security fix support on python 3.6.8 :-) which should be enough for me for the most part (we run RHEL, but haven't made the move to AAP yet...). But we'll keep the situation on the radar and plan ahead for adopting RHEL9 accordingly.

Thanks again, you guys are awesome <3
